Sources/CartonHelpers/DiagnosticsParser.swift

```diff
@@ -0,0 +1,225 @@
+        case .warning: return "[43;1m" // bright yellow background
+        case .note: return "[7m" // reversed
+        }
+      }
+    }
+  }
+
+  fileprivate static let highlighter = SyntaxHighlighter(format: TerminalOutputFormat())
+
+  func parse(_ output: String, _ terminal: InteractiveWriter) {
+    let lines = output.split(separator: "\n")
+    var lineIdx = 0
+
+    var diagnostics = [String.SubSequence:[CustomDiagnostic]]()
+
+    var currFile: String.SubSequence?
+    var fileMessages = [CustomDiagnostic]()
+
+    while lineIdx < lines.count {
+      let line = lines[lineIdx]
+      if let file = line.matches(regex: Regex.enterFile) {
+        if let currFile = currFile {
+          diagnostics[currFile] = fileMessages
+        }
+        currFile = file
+        fileMessages = []
+      } else if let currFile = currFile {
+        if let message = line.matches(regex: Regex.line) {
+          let components = message.split(separator: ":")
+          if components.count > 3 {
+            lineIdx += 1
+            let file = line.replacingOccurrences(of: message, with: "")
+            guard file.split(separator: "/").last?.replacingOccurrences(of: ":", with: "") == String(currFile) else { continue }
+            fileMessages.append(
+              .init(
+                kind: CustomDiagnostic.Kind(rawValue: String(components[2].trimmingCharacters(in: .whitespaces))) ?? .note,
+                file: file,
+                line: components[0],
+                char: components[1],
+                code: String(lines[lineIdx]),
+                message: components[3]
+              )
+            )
+          }
+        }
+      } else {
+        terminal.write(String(line) + "\n", inColor: .cyan)
+      }
+      lineIdx += 1
+    }
+    if let currFile = currFile {
+      diagnostics[currFile] = fileMessages
+    }
+
+    for (file, messages) in diagnostics.sorted(by: { $0.key < $1.key }) {
+      guard messages.count > 0 else { continue }
+      terminal.write("\(" \(file) ", color: "[1m", "[7m")") // bold, reversed
+      terminal.write(" \(messages.first!.file)\(messages.first!.line)\n\n", inColor: .grey)
+      // Group messages that occur on sequential lines to provie a more readable output
+      var groupedMessages = [[CustomDiagnostic]]()
+      for message in messages {
+        if let lastLineStr = groupedMessages.last?.last?.line,
+           let lastLine = Int(lastLineStr),
+           let line = Int(message.line),
+           lastLine == line - 1 || lastLine == line {
+          groupedMessages[groupedMessages.count - 1].append(message)
+        } else {
+          groupedMessages.append([message])
+        }
+      }
+      for messages in groupedMessages {
+        // Output the diagnostic message
+        for message in messages {
+          terminal.write("  \(" \(message.kind.rawValue.uppercased()) ", color: message.kind.color, "[37;1m") \(message.message)\n") // 37;1: bright white
```

Comment:
<!--
  1 failure:  Line should be 10...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: Sources/CartonHelpers/DiagnosticsParser.swift;
  Line: 183;
-->
- :no_entry_sign: Line should be 100 characters or less: currently 154 characters (`line_length`)